web/packages/teleterm/src/ui/ConnectMyComputer/connectMyComputerContext.tsx

```diff
@@ -283,7 +283,7 @@ export const ConnectMyComputerContextProvider: FC<{
     const shouldAutoStartAgent =
       isAgentConfigured &&
       canUse &&
-      !isNonCompatibleAgent &&
+      isAgentCompatible &&
```

Comment:
Good catch! We should definitely show it on the icon.

Reply:
I realized that in an ideal world, the error indicator would only show up if autostart was prevented due to incompatibility issues. Otherwise if you configure the agent, stop it and compatibility issues occur in the future, you're going to see the error indicator even though you're not actively using the agent.

Anyways, I think we can keep it as it is, at least it'll nudge those users towards an upgrade of Connect.

Reply:
Yes, it would be better, but IMO it's still not bad, at least users will be more aware that they are using an incompatible version.

Reply:
I ran into another issue with the indicator.

`isAgentCompatible` looks at `rootCluster.proxyVersion`, but this information is not immediately available. So if I configure my agent and then restart the app, the indicator will show a warning color until the app fetches detailed information about the cluster.

I think we should be able to fix this by replacing `isAgentCompatible` with `agentCompatibility` that can be one of three values: compatible, incompatible, unknown. I'll attempt to fix it next week.

---

Also, during the review I forgot that differentiating between success & error states with just color is usually not enough for users with red-green confusion.

https://github.com/gravitational/teleport/assets/27113/fc3c170d-e984-4801-82ba-8e1e43b4c71a

Reply:
Ah, I didn't think of it 😞

>I think we should be able to fix this by replacing isAgentCompatible with agentCompatibility that can be one of three values: compatible, incompatible, unknown.

👍

As for colors, probably we should use an icon? I can take a look at it.

Reply:
Yeah, an icon would be a good solution.